### PR TITLE
refactor(RingTheory/Trace): Define `TraceForm` with `LinearMap.BilinForm`

### PR DESCRIPTION
**Pull request by mans0954**
Redefine `traceForm : BilinForm R S` using `LinearMap.BilinForm` instead of `BilinForm`.

Contributes to  #10553

---

- [x] depends on: #11032

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Trace.lean [hunk 1]
```diff
@@ -190,8 +193,8 @@ variable (R S)
 /-- The `traceForm` maps `x y : S` to the trace of `x * y`.
 It is a symmetric bilinear form and is nondegenerate if the extension is separable. -/
 noncomputable def traceForm : BilinForm R S :=
-  -- Porting note: dot notation `().toBilin` does not work anymore.
-  LinearMap.toBilin (LinearMap.compr₂ (lmul R S).toLinearMap (trace R S))
+-- Porting note: dot notation `().toBilin` does not work anymore.
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Trace.lean:196]:
```suggestion
```
This does not make sense anymore.

File: Mathlib/RingTheory/Trace.lean [hunk 2]
```diff
@@ -622,15 +623,15 @@ theorem det_traceForm_ne_zero [IsSeparable K L] [DecidableEq ι] (b : Basis ι K
...
 #align det_trace_form_ne_zero det_traceForm_ne_zero
 
 variable (K L)
 
-theorem traceForm_nondegenerate [FiniteDimensional K L] [IsSeparable K L] :
-    (traceForm K L).Nondegenerate :=
-  BilinForm.nondegenerate_of_det_ne_zero (traceForm K L) _
-    (det_traceForm_ne_zero (FiniteDimensional.finBasis K L))
-#align trace_form_nondegenerate traceForm_nondegenerate
+theorem traceForm_separatingLeft [FiniteDimensional K L] [IsSeparable K L] :
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Trace.lean:631]: Why erasing `traceForm_nondegenerate`? Even if `traceForm_separatingLeft` is stronger we can have both (as a lot of people will look for nondegeneracy).

**Comment by riccardobrasca** [comment]
The description of the PR is wrong, right? You use `LinearMap.compr₂`.

File: Mathlib/RingTheory/Trace.lean (hunk 1 above)
Comment (mans0954) [reply Mathlib/RingTheory/Trace.lean]: `traceForm` got redefined in #12078, but the porting note remains.

**Comment by mans0954** [comment]
@eric-wieser do you think I should close this PR without merging? I'm not sure that it's useful now #11278 has been merged.

File: Mathlib/RingTheory/Trace.lean (hunk 1 above)
Comment (riccardobrasca) [reply Mathlib/RingTheory/Trace.lean]: Well, I guess it's time to remove it.

Comment (mans0954) [reply Mathlib/RingTheory/Trace.lean]: #12489

**Comment by mans0954** [comment]
Closing this, for now at least, as I don't think it helps us move things forwards.

**Pull request closed without merging**